File: BackEnd/backend/src/main/java/com/fitapp/backend/infrastructure/persistence/entity/enums/ExerciseType.java
package com.fitapp.backend.infrastructure.persistence.entity.enums;

public enum ExerciseType {
    SIMPLE,          // Solo un parámetro principal
    WEIGHTED,        // Con peso
    TIMED,           // Con tiempo
    MIXED,           // Múltiples parámetros (peso + tiempo)
    BODYWEIGHT,      // Peso corporal
    DISTANCE,        // Distancia
    REPETITION,      // Solo repeticiones
    DURATION,        // Solo duración
    CIRCUIT,         // Circuito
    AMRAP,           // As Many Rounds As Possible
    EMOM,            // Every Minute On the Minute
    TABATA           // Intervalos específicos
}

File: FrontEnd/app/src/main/java/com/fitapp/appfit/feature/exercise/model/exercise/response/ExerciseType.kt
package com.fitapp.appfit.feature.exercise.model.exercise.response

import com.google.gson.annotations.SerializedName

enum class ExerciseType {
    @SerializedName("SIMPLE")
    SIMPLE,

    @SerializedName("WEIGHTED")
    WEIGHTED,

    @SerializedName("TIMED")
    TIMED,

    @SerializedName("MIXED")
    MIXED,

    @SerializedName("BODYWEIGHT")
    BODYWEIGHT,

    @SerializedName("DISTANCE")
    DISTANCE,

    @SerializedName("REPETITION")
    REPETITION,

    @SerializedName("DURATION")
    DURATION,

    @SerializedName("CIRCUIT")
    CIRCUIT,

    @SerializedName("AMRAP")
    AMRAP,

    @SerializedName("EMOM")
    EMOM,

    @SerializedName("TABATA")
    TABATA
}